### ISSUE TITLE
Incorrect shelving dates after goodreads import

### ISSUE THREAD

**Issue by emiljacobsen** [bookwyrm-social/bookwyrm#1174]
**Describe the bug**
The shelving dates are all wrong after importing my library from goodreads. For example, looking at goodreads, I can see that I added Slaughterhouse Five on July 26, 2016. In bookwyrm on the other hand, the shelving date is October 30, 2020. In general, it seems all the shelving dates on bookwyrm are quite recent (but many still prior to the date I joined the website). I see no other pattern to what dates were chosen.

**To Reproduce**
1. Import library from goodreads.
2. Compare shelving dates in bookwyrm to added dates in goodreads.
3. Note the difference.

**Expected behavior**
I expected the dates to either coincide, or for all the bookwyrm shelving dates be the date I imported the library.

**Screenshots**
![bookwyrm-sh5](https://user-images.githubusercontent.com/1376211/121950228-a50fed80-cd59-11eb-9947-2c67c6399dfe.png)
![goodreads-sh5](https://user-images.githubusercontent.com/1376211/121950234-a5a88400-cd59-11eb-8e02-1252d45d5aea.png)

**Comment by mouse-reeve** [comment]
Have you added read dates retroactively to goodreads? For example, is it possible that you read _Slaughterhouse Five_ in 2016, but added it to goodreads in 2020?

Sometimes goodreads doesn't include the reading start date, and we use the date added instead.

**Comment by emiljacobsen** [comment]
No, sorry, I should have been more specific. The book has no read dates, it's only ever been on the 'to read' shelves, and I can't associate any other dates to it than July 26. In fact, I was only looking at books in this situation when I noticed the discrepancy.

I'm now comparing dates for books in my 'read' shelf, and these too differ between the sites. Here's an interesting example: A People's History of the United States:

On goodreads:
- added July 26, 2016
- started reading July 26, 2016
- finished reading March 9, 2017

On bookwyrm:
- shelved October 30, 2020
- started reading July 26, 2016
- finished reading March 9, 2017

Note: not all the books have these two dates 26.6.2016 and 30.10.2020... I just happened to pick two that did. It seems to not always be the case. On goodreads, I added Masters of Doom and The Waves on March 1, 2017. On bookwyrm, they're shelved November 8, 2020 and January 4, 2021, respectively.

**Comment by mouse-reeve** [comment]
Would you be comfortable attaching the csv here or emailing it to me (mousereeve@riseup.net)?

**Comment by emiljacobsen** [comment]
Sent by email!

**Comment by cincodenada** [comment]
I've been looking into what I think is the same issue: ran an import and it looks like the "shelved" date is taken from the `created_date` field on the ShelfBook, which is not taken from the CSV at all, but rather is when I ran the import.

Unfortunately the `created_date` field, being `auto_now_add`, is essentially immutable - it's set when we create the record, and we can't override it.

My proposal is to add a `shelved_date` field to the ShelfBook table that would default to the current date/time, but for imports would be overrideable. I've started in on this work, so I have a PR half-finished locally, but I wanted to see if this seems like a good solution before I get too far along :)
